src/stats/dendrogram.jl

```diff
@@ -0,0 +1,123 @@
+"""
+    dendrogram(x, y; kwargs...)
+
+Draw a [dendrogram](https://en.wikipedia.org/wiki/Dendrogram),
+with leaf nodes specified by `x` and `y` coordinates,
+and parent nodes identified by `merges`.
+
+# Arguments
+- `x`: x positions of leaf nodes
+- `y`: y positions of leaf nodes (default = 0)
+# Keywords
+- `merges`: specifies connections between nodes (see below)
+- `treestyle`: one of `:??`, `:??` 
+"""
+@recipe(Dendrogram, x, y) do scene
+    Theme(
+        weights = automatic,
+        color = theme(scene, :patchcolor),
+        colormap = theme(scene, :colormap),
+        colorrange = automatic,
+        orientation = :vertical,
+        strokecolor = theme(scene, :patchstrokecolor),
+        strokewidth = theme(scene, :patchstrokewidth),
+        cycle = [:color => :patchcolor],
+        inspectable = theme(scene, :inspectable)
+    )
+end
+
+conversion_trait(x::Type{<:Dendrogram}) = SampleBased() #??
```

Comment:
I don't think we have this kind of trait - if you want to make one, you can define it as `struct SampleBased <: MakieCore.ConversionTrait end`.

You don't _have_ to define this - what kind of arguments could `dendrogram` expect?  Could there be other recipes to which this would be common?

Reply:
I copied that from BarPlot - no idea what it does 😅

Reply:
Oh, my bad - was looking at MakieCore.  It does seem to exist :D.

`conversion_trait` is basically a trait system for plots, where e.g. `lines`, `scatter`, and `poly` all share the `PointBased` trait - this enables people to write converts for their data types dispatching on PointBased, and thereby hook that in to the conversion pipeline without having to specify a specific plot type.